oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MissingLastRevSeekerTest.java

```diff
@@ -249,7 +251,7 @@ public void dispose() {
         }
         dns.runBackgroundOperations();
         //seeker should return only non split documents
-        int docs = Iterables.size(seeker.getCandidates(0));
+        int docs = Iterables.size(seeker.getCandidates(Objects.requireNonNull(dns.getRoot().getLastRevision().getRevision(dns.getClusterId())).getTimestamp()));
```

Comment:
Is this related to the flakyness?

Reply:
Yes. I removed 0 and replaced it with proper revision version.

Reply:
Ok.

a) Could you put a short description of what was wrong into the ticket?

b) I would recommend that @egli, @rishabhdaim or @Joscorbe confirm that his is the correct change.

Reply:
`0` was used to fetch all the documents. Using the timestamp of lastRevision would result in skipping some documents.

 I would revert the change.

Reply:
I was able to reproduce this test failure on my local. The root cause the is presence of document with id `1:/test` in the `RDB-H2`.

Once I did `mvn clean install` this problem went away.
IMO, it is because of some earlier test did didn't cleanup properly rather than test.

Reply:
Yes. That is the reason that I modified the timestamp to return only the documents needed, not all documents.
@reschke we can have inconsistencies retrieving all docs from RDB, but retrieving only the ones the this test added should solve the issue and future issues that could appear regarding this test. What do you think?

Reply:
I think I found the Root cause.

These tests are failing due to the `CompressedDocumentPropertyStateTest` test not cleaning up properly.
On my local, it is causing either Mongo or RDB-H2 to fail.

`CompressedDocumentPropertyStateTest` creates the problematic node `1:/test` causing test failures.

`CompressedDocumentPropertyStateTest` has been added recently which explains the recent failures of `MissingLastRevSeekerTest`.

@ionutzpi could you please look into this?
cc @Joscorbe @reschke

Reply:
Yes. I added CompressedDocumentPropertyStateTest changes to solve this issue. Julian told that documentfixturestore should close when closing nodestore(seems that this is not true).

Reply:
Calling `fixture.dispose();` after disposing of `nodeStore` [1] would fix this problem for `Mongo`.

[1] https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/CompressedDocumentPropertyStateTest.java#L240

Reply:
It is not removing for RDB, I dont know the reason yet.

Reply:
@rishabhdaim if I add removeMe.add("1:/test"); after assert it will fix the issue. Do you agree with this change?